src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs

```diff
@@ -0,0 +1,208 @@
+                }
+            }
+
+            span.ResetStartTime();
+            return new CallTargetState(scope);
+        }
+
+        /// <summary>
+        /// OnAsyncMethodEnd callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TReturn">Type of the return value</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="returnValue">Return value</param>
+        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+        /// <param name="state">Calltarget state value</param>
+        /// <returns>A response value, in an async scenario will be T of Task of T</returns>
+        public static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, CallTargetState state)
+            where TTarget : ITestMethodRunner
+        {
+            Scope scope = state.Scope;
+            if (scope != null)
+            {
+                Array returnValueArray = returnValue as Array;
+                if (returnValueArray.Length == 1)
+                {
+                    object unitTestResultObject = returnValueArray.GetValue(0);
+                    if (unitTestResultObject != null)
+                    {
+                        UnitTestResultStruct unitTestResult = unitTestResultObject.As<UnitTestResultStruct>();
+
+                        switch (unitTestResult.Outcome)
+                        {
+                            case UnitTestResultOutcome.Error:
+                            case UnitTestResultOutcome.Failed:
+                            case UnitTestResultOutcome.NotFound:
+                            case UnitTestResultOutcome.Timeout:
+                                scope.Span.SetTag(TestTags.Status, TestTags.StatusFail);
+                                scope.Span.Error = true;
+                                scope.Span.SetTag(Tags.ErrorMsg, unitTestResult.ErrorMessage);
+                                scope.Span.SetTag(Tags.ErrorStack, unitTestResult.ErrorStackTrace);
+                                break;
+                            case UnitTestResultOutcome.Inconclusive:
+                            case UnitTestResultOutcome.NotRunnable:
+                            case UnitTestResultOutcome.Ignored:
+                                scope.Span.SetTag(TestTags.Status, TestTags.StatusSkip);
+                                scope.Span.SetTag(TestTags.SkipReason, unitTestResult.ErrorMessage);
+                                break;
+                            case UnitTestResultOutcome.Passed:
+                                scope.Span.SetTag(TestTags.Status, TestTags.StatusPass);
+                                break;
+                        }
+                    }
+                }
+
+                scope.Dispose();
```

Comment:
Is there a scenario in which the exception could be populated? Should we call `scope.DisposeWithException(exception)` to be on the safe side?

Reply:
Nevermind, it appears that an exception will be set via the other integration method